cmd/clusterctl/client/alpha/rollout_viewer_test.go

```diff
@@ -0,0 +1,338 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package alpha
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/internal/test"
+)
+
+func captureStdout(fnc func()) string {
+	r, w, _ := os.Pipe()
+
+	stdout := os.Stdout
+	defer func() {
+		os.Stdout = stdout
+	}()
+	os.Stdout = w
+
+	fnc()
+	_ = w.Close()
+
+	var buf bytes.Buffer
+	_, _ = io.Copy(&buf, r)
```

Comment:
Do we need to have the `_, _` assignment here?

Reply:
Otherwise the following linter error happens, but ignoring err is certainly bad practice... I changed it to return an error. Thank you for pointing it out.

```bash
cmd/clusterctl/client/alpha/rollout_viewer_test.go:48:9: Error return value of `io.Copy` is not checked (errcheck)
        io.Copy(&buf, r)
```